src/nunavut/lang/cpp/templates/deserialization.j2

```diff
@@ -199,8 +199,11 @@
 {% endif %}
     {# GENERAL CASE #}
     {% set ref_index = 'index'|to_template_unique_name %}
-        for (size_t {{ ref_index }} = 0U; {{ ref_index }} < {{ reference }}.size(); ++{{ ref_index }})
+        for (size_t {{ ref_index }} = 0U; {{ ref_index }} < {{ ref_size }}; ++{{ ref_index }})
         {
+            // TODO This is terribly inefficient. We need to completely refactor this template to use C++ emplace and
```

Comment:
This is crap but part of the "experimental" state of C++ generation. We'll fix this before calling it GA.